tests/ui/resolve/issue-56593-2.rs

```diff
@@ -0,0 +1,26 @@
+// check-pass
+
+use thing::*;
+
+#[derive(Debug)]
+pub enum Thing {
+    Foo,
+}
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+
+    #[test]
+    fn test_thing() {
+        let thing = Thing::Foo;
```

Comment:
break change:
- before: `root::Thing` refers to `root::thing::Thing`
- after this PR: `root::Thing` refers to `root::Thing`